src/Sulu/Component/Webspace/Analyzer/Attributes/ParameterRequestProcessor.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Webspace\Analyzer\Attributes;
+
+use Sulu\Component\Webspace\Manager\WebspaceManagerInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Takes the parameters from the requests, and tries to find a suiting portal. It checks for the parameters _locale and
+ * _portal. Based on these it will load the best matching portal information, and uses it to analyze the request.
+ */
+class ParameterRequestProcessor extends AbstractPortalInformationRequestProcessor
+{
+    /**
+     * @var WebspaceManagerInterface
+     */
+    private $webspaceManager;
+
+    /**
+     * @var string
+     */
+    private $environment;
+
+    public function __construct(
+        WebspaceManagerInterface $webspaceManager,
+        $environment
+    ) {
+        $this->webspaceManager = $webspaceManager;
+        $this->environment = $environment;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(Request $request, RequestAttributes $requestAttributes)
+    {
+        if (!$request->get('_locale') && !$request->get('_portal')) {
+            return new RequestAttributes();
+        }
+
+        $portalInformations = $this->webspaceManager->findPortalInformationsByPortalKeyAndLocale(
+            $request->get('_portal'),
+            $request->get('_locale'),
+            $this->environment
+        );
+
+        if (!$portalInformations) {
+            return new RequestAttributes();
+        }
+
+        return $this->processPortalInformation($request, reset($portalInformations));
```

Comment:
I've realized that some of the `RequestProcessor`s are inheriting from the same class, and couple the logic in here. I would suggest to let the `WebsiteRequestProcessor` and the `CustomUrlRequestProcessor` only implement the `RequestProcessorInterface`, and let them only set the `portalInformation` request attribute. Then there could be another `RequestProcessor`, which is only responsible for adding all the other information based on the `PortalInformation` to the Request. This way the responsibilities are better separated, and all the stuff done in `processPortalInformation` would only be executed once (at the very end).

@wachterjohannes @chirimoya WDYT?

I could do that now, but I don't know if it is too much for a hotfix, although we are already changing quite a lot here anyway... Otherwise I would just create an issue, and maybe do that on develop soonish.